### PR TITLE
Create mediachain-rfc-3.md

### PR DESCRIPTION
**Pull request by autoencoder**
- Addressed suggested changes - Missing any?
- Added lifecycle descriptions for training data & hyper-parameters.
- Rewrote & expanded most sections.

TODO:
- Describe other lifecycles?
- Minor normalization of style, word tenses, phrasing.
- Exact APIs will be in flux as we develop them.

### CODE REVIEW

File: rfc/drafts/mediachain-rfc-3.md [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+The `Ingestion` subsystem accepts media works from the `Transactors` via the `Clients`, performs basic attack
+mitigation and media prioritization, and then processes the media objects into image feature descriptors. These
+feature descriptors, along with the media IDs are then inserted into the `KNN Index`, which will be queried by
+the `Search` and `Dedupe` systems.
+
+User interfaces to the `Ingestion`  subsystem will in coordination with the `Dedupe` subsystem, to alert users if they are
+about to create duplicate entries for the same work in the MediaChain blockchain, or allow users to provide a negative training
+signal to the deduplication subsystem in the case where media works have been incorrectly flagged as duplicates.
+
+In the case when an exact match of media content, which was not caught by the MediaChain core - for example images files with the same
```
Comment (parkan) [inline rfc/drafts/mediachain-rfc-3.md:76]: Should clear up how this works in relation to the rest of the system

Comment (parkan) [inline rfc/drafts/mediachain-rfc-3.md:76]: (not necessarily in scope here, just need to work through it)

File: rfc/drafts/mediachain-rfc-3.md [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+## Search Subsystem
+
+The `Search` system is exposed to end-users via a REST/JSON API. The address of the REST API will be communicated to end-users via the
+Mediachain Core. This API allows users to input a query or media ID, and receive a ranked list of most semantically-relevant media IDs.
+
+
+ Method | REST Call     | Description                                       | Returns
+ -------|---------------|---------------------------------------------------|---------------------------------------------------
+ POST   | `/search`     | Search by base64-encoded image or image ID.       | List of image IDs, possibly with relevancy scores.
+ POST   | `/distance`   | Calculate metric-space distance between 2 images. | Numeric distance between 0.0 and 1.0.
```
Comment (parkan) [inline rfc/drafts/mediachain-rfc-3.md:93]: cosine distance? "metric-space" doesn't imply boundedness

**Comment by parkan** [comment]
lgtm! We can definitely refine this further but it's in a great place for now

**Comment by parkan** [comment]
One thing I'd like to see in more detail is a detailed ingester logic description (outside the indexer per se), something like
- indexer receives a `JournalEntry` from transactors
  - if `CanonicalEntry`, ...
  - if `ChainEntry` -> follow new `chain` head
    - if new head is `ArtefactUpdateCell` -> some actions depending on whether it has thumbnail, keyword, etc fields or a sourced from ref etc
    - if new head is `ArtefactCreationCell` -> ...

File: rfc/drafts/mediachain-rfc-3.md (hunk 2 above)
Comment (autoencoder) [inline rfc/drafts/mediachain-rfc-3.md:93]: Yes, that line got quite jumbled... In fact, let's rather have a similarity function there instead of a distance, and we'll change the description to "Calculate similarity score for a pair of images".

File: rfc/drafts/mediachain-rfc-3.md [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+     |                                 ^
++----v-----+  +--------------------+   |
+|user query+->|run query on        +->-/
++----------+  |for search / dedupe |
+              |system based on     |
+              |unsupervised,       |
+              |engineered features.|
+              +--------------------+
+```
+
```
Comment (autoencoder) [inline rfc/drafts/mediachain-rfc-3.md:317]: Above is meant to say (manual tuning) instead of (no tuning).

**Comment by parkan** [comment]
gonna merge this
